apps/etherscan/src/app/utils/verify.ts

```diff
@@ -85,13 +87,20 @@ export const verify = async (
         module: "contract", // Do not change
         action: "verifysourcecode", // Do not change
         codeformat: "solidity-standard-json-input",
-        contractaddress: contractAddress, // Contract Address starts with 0x...
         sourceCode: JSON.stringify(jsonInput),
         contractname: fileName + ':' + contractName,
         compilerversion: `v${contractMetadataParsed.compiler.version}`, // see http://etherscan.io/solcversions for list of support versions
         constructorArguements: contractArgumentsParam ? contractArgumentsParam.replace('0x', '') : '', // if applicable
       }
 
+      if (isProxyContract) {
+        data.action = "verifyproxycontract"
+        data.expectedimplementation = expectedImplAddress
```

Comment:
you don't seem to use that `expectedimplementation` anywhere, what is that for?

Reply:
It is an optional parameter and used in the API. See: https://docs.etherscan.io/api-endpoints/contracts#verify-proxy-contract

Reply:
ok where is it used for the api call?